bolt/include/bolt/Core/BinaryFunction.h

```diff
@@ -817,6 +820,12 @@ class BinaryFunction {
     return nullptr;
   }
 
+  /// Indicates if the function is safe to fold.
+  bool isSafeToICF() const { return IsSafeToICF; }
+
+  /// Sets the function is not safe to fold.
+  void setUnsafeICF() { IsSafeToICF = false; }
```

Comment:
A better interface might be `setSafeICF(bool isSafe)`, maybe. I'll let @maksfb comment.